build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/extractor/GoZipBallStreamer.java

```diff
@@ -238,29 +244,59 @@ private boolean isSubFolderOfAnotherModule(String moduleRootDir, String director
      * @return True if the entry is a go.mod file which doesn't belong to the project we are packing
      */
     private boolean isSubModule(String entryName) {
-        if (entryName.endsWith(MOD_FILE)) {
+        if (entryName.endsWith(MOD_FILE_PATH)) {
             if (shouldPackSubModule()) {
-                return (!entryName.substring(entryName.indexOf('/') + 1).endsWith(subModuleName + MOD_FILE));
+                return (!entryName.substring(entryName.indexOf('/') + 1).endsWith(subModuleName + MOD_FILE_PATH));
             } else {
-                return entryName.substring(entryName.indexOf('/') + 1).endsWith(subModuleName + MOD_FILE);
+                return !entryName.substring(entryName.indexOf('/') + 1).equals(MOD_FILE);
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @param majorVersion - major version of a compatible v2 project
+     * @return true, if the go.mod file in the root specify a path of submodule of the given major version
+     */
+    private boolean hasRootModFileOfCompatibleModuleFromV2(String majorVersion) {
+        Enumeration<? extends ZipArchiveEntry> entries = zipFile.getEntries();
+        if (entries.hasMoreElements()) {
+            ZipArchiveEntry zipEntry = entries.nextElement();
+            ZipArchiveEntry modEntry = zipFile.getEntry(getFirstPathElement(zipEntry.getName()) + MOD_FILE_PATH);
+            if (modEntry == null) {
+                modEntry = zipFile.getEntry(MOD_FILE);
+            }
+            if (modEntry != null) {
+                try (InputStream inputStream = zipFile.getInputStream(modEntry)) {
+                    String modFileContent = IOUtils.toString(inputStream, Charset.defaultCharset());
+                    return modFileContent.substring(0, modFileContent.indexOf("\n")).endsWith("/" + majorVersion);
```

Comment:
I'm not sure it will work on Windows, whereby in some cases the new line character is `\r\n`.
Let's trim the string before calling `endsWith`.
Also - if we replace `substring` with `StringUtils.substringBefore`, the code wouldn't crash on empty go.mod files.